File: RGB거리.cpp
#include<iostream>
#include<vector>
#include<queue>
using namespace std;

int n;
int arr[1001][3] = {};
int dp[1001][3] = {};
int main() {
	cin >> n;
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < 3; j++) {
			cin >> arr[i][j];
		}
	}

	dp[0][0] = arr[0][0], dp[0][1] = arr[0][1], dp[0][2] = arr[0][2];
	for (int i = 1; i < n; i++) {
		for (int j = 0; j < 3; j++) {
			dp[i][j] = min(dp[i - 1][(j + 1) % 3], dp[i - 1][(j + 2) % 3]) + arr[i][j];
		}
	}

	cout << min(min(dp[n - 1][0], dp[n - 1][1]), dp[n - 1][2]);
	return 0;
}

File: PS-DP/RGB거리.py
import sys
input = sys.stdin.readline
n = int(input())
rgb = []
ret = [[1000001] * 3 for _ in range(n)]
for i in range(n):
    arr = list(map(int,input().split()))
    rgb.append(arr)
    
ret[0] = rgb[0]
for i in range(1,n):
    for j in range(3):
        ret[i][j] = min(ret[i-1][(j+1) % 3], ret[i-1][(j+2) % 3]) + rgb[i][j]

print(min(ret[n-1][0],ret[n-1][1],ret[n-1][2]))